modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -242,99 +241,147 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
+     * @param excludeAttributes Exclude attributes
+     * @return SCIM response
+     */
+    @Override
+    public SCIMResponse listWithGET(UserManager userManager, String filter, Integer startIndexInt, Integer countInt,
+            String sortBy, String sortOrder, String domainName, String attributes, String excludeAttributes) {
 
-                if (tempList == null) {
-                    tempList = Collections.emptyList();
-                }
+        FilterTreeManager filterTreeManager;
+        try {
+            Integer count = ResourceManagerUtil.processCount(countInt);
+            Integer startIndex = ResourceManagerUtil.processStartIndex(startIndexInt);
 
-                try {
-                    totalResults = (int) tempList.get(0);
-                    tempList.remove(0);
-                } catch (IndexOutOfBoundsException e) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Group result list is empty.");
-                    }
-                    totalResults = tempList.size();
-                } catch (ClassCastException ex) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Parse error while getting the group result count. Setting result count as: " +
-                                     tempList.size(), ex);
-                    }
-                    totalResults = tempList.size();
-                }
-                returnedGroups = tempList;
+            // Resolving sorting order.
+            sortOrder = resolveSortOrder(sortOrder, sortBy);
 
-                for (Object group : returnedGroups) {
-                    //perform service provider side validation.
-                    ServerSideValidator.validateRetrievedSCIMObjectInList((Group) group, SCIMSchemaDefinitions
-                                    .SCIM_GROUP_SCHEMA,
-                            attributes, excludeAttributes);
-                }
-                //create a listed resource object out of the returned groups list.
-                ListedResource listedResource = createListedResource(returnedGroups, startIndex, totalResults);
-                //convert the listed resource into specific format.
-                String encodedListedResource = encoder.encodeSCIMObject(listedResource);
-                //if there are any http headers to be added in the response header.
-                Map<String, String> responseHeaders = new HashMap<String, String>();
-                responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
-                return new SCIMResponse(ResponseCodeConstants.CODE_OK, encodedListedResource, responseHeaders);
+            // Unless configured returns core-user schema or else returns extended user schema.
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
 
+            //// Build node for filtering.
```

Comment:
Shall we format this comment // ?

Reply:
fixed: https://github.com/wso2/charon/pull/263/commits/61f4e744f53695f16e5ad6c9496f369ff77101f8